src/LanguageServer/Impl/Implementation/Server.Editor.cs

```diff
@@ -42,8 +42,14 @@ public sealed partial class Server {
                 var result = _completionSource.GetCompletions(analysis, @params.position);
                 res.items = result?.Completions?.ToArray() ?? Array.Empty<CompletionItem>();
 
-                await InvokeExtensionsAsync((ext, token)
-                    => (ext as ICompletionExtension)?.HandleCompletionAsync(analysis, @params.position, res.items.OfType<CompletionItemEx>().ToArray(), cancellationToken), cancellationToken);
+                await InvokeExtensionsAsync(async (ext, token)
+                    => {
+                        if (ext is ICompletionExtension2) {
```

Comment:
A pattern matching switch case would probably be better here, rather checking `is` then `as`. (Or even `if (ext is ICompletionExtension e) { await e.Handle...`)

Reply:
Refactored. Thx.